docs/content/v0.20/architecture.md

```diff
@@ -292,42 +307,43 @@ follows:
-          find the matching clusters using the label selector expression
-          for clusters.
-        - If there are matching clusters, add the names of the cluster
-          to a hashmap setting the name of the cluster as a key. Cluster
-          groups from different placements are merged together.
-        - If any of the matched Placements has ` WantSingletonReportedState`
-          set to true, clusters are sorted in alphanumerical order and
-          only the first cluster is selected for delivery. Note that setting 
-          `WantSingletonReportedState` in one of the placements that 
-          matches the object affects the behavior for all matching placements, 
-          those with matching clusters and those w/o matching clusters. 
-    -  If there are no matching clusters, the worker returns without
-        actions and is ready to process other events from the queue.
-    -  If there are matching clusters:
-        - Wraps the object into a ManifestWork
-        - Adds a label for each matched Placement to the ManifestWork that 
-          is used to track the placement that caused the object to be 
-          delivered to one or more clusters. The label contains both the 
-          placement name (note that placement is cluster-scoped) and the WDS name. 
-          This way, when a placement is deleted or updated it is possible to 
-          locate the associated ManifestWorks for deletion or label removal 
-          (if more than one label is present, as other placements may “own” the object).
-        - For each cluster:
-          -  Sets the manifest namespace == name of the cluster
-          -  Uses the client connected to the ITS to do a server-side
-             apply patch of the manifest into the namespace.
-          - At this time there is only one field manager name for the server-side
-            apply (`kubestellar`) but the name should include also the WDS
-            name to allow detecting conflicts when two different field managers
-            try to patch a manifest with the same name.   
-        - Worker returns and is ready to pick other keys from the queue.
+    -  Gets the lister for BindingPolicy objects and list all binding-policies
+    -  Iterates on all binding-policies, and for each of them:
+        - Evaluates whether the object matches the downsync selection 
+          criteria in the `BindingPolicy`.
+        - If there is a match:
+          - Notes the object as a match for the in-memory representation of the relevant `Binding`.
+            - Enqueues the representation of the relevant `Binding` for syncing if changed. 
+          - Lists ManagedClusters and finds the matching clusters using the label selector expression for clusters.
+            - If there are matching clusters, adds the names of the cluster
+            to a hashmap setting the name of the cluster as a key. Cluster
+            groups from different bindind policies are merged together.
```

Comment:
FYI: Should be indented a little more. Not a functional problem, it seems to work anyway, but would be clearer to read.

Reply:
fixed